main.cpp

```diff
@@ -19,13 +19,34 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter &rewriter) : rewriter_(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *castExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");

```

Comment:
What if `castExpr` is a null pointer? How to protect the program from crashing in such cases?

Reply:
To protect the program from crashing in the case when `castExpr` is a null pointer, can add the following check: 
`if (castExpr == nullptr) return;`